code/modules/surgery/organs/organ.dm

```diff
@@ -248,13 +248,34 @@ var/list/organ_cache = list()
 		return
 	switch (severity)
 		if (1.0)
-			take_damage(0,20)
+			take_damage(20,0)
 			return
 		if (2.0)
-			take_damage(0,7)
+			take_damage(7,0)
 			return
 		if(3.0)
-			take_damage(0,3)
+			take_damage(3,0)
+
+/obj/item/organ/internal/emp_act(severity)
+	if(!robotic)
+		return
+	if(ORGAN_ROBOT)
```

Comment:
You're essentially checking `if(128)` on this line. Defines are static values, this doesn't check anything about the organ itself. Tiger is probably right here, although I know there's something stupid about organs where there are *two* variables that are set on robotic organs or something like that.